pecos/xmc/xtransformer/matcher.py

```diff
@@ -1316,13 +1362,23 @@ def train(
             val_prob.X_text = val_tensors
 
         bootstrapping = kwargs.get("bootstrapping", None)
-        if bootstrapping is not None:
-            init_encoder, init_embeddings = bootstrapping
+        if train_params.bootstrap_method is not None and bootstrapping is not None:
+            init_encoder, init_embeddings, prev_head = bootstrapping
             matcher.text_encoder.init_from(init_encoder)
             LOGGER.info("Initialized transformer text_encoder form given text_encoder!")
-            if train_params.do_text_model_bootstrap and init_embeddings is not None:
-                matcher.text_model.bootstrap(init_embeddings, prob.Y, C=prob.C, M=prob.M)
-                LOGGER.info("Initialized transformer text_model form given embeddings!")
+            if train_params.bootstrap_method == "linear" and init_embeddings is not None:
+                bootstrap_prob = MLProblem(
+                    init_embeddings,
+                    prob.Y,
+                    C=prob.C if prob.M is not None else None,
+                    M=prob.M,
+                    R=prob.Y if "weighted" in train_params.loss_function else None,
+                )
+                matcher.text_model.bootstrap(bootstrap_prob)
+                LOGGER.info("Initialized transformer text_model with xlinear!")
+            elif train_params.bootstrap_method == "inherit":
```

Comment:
In practice, do we ever use this "inherit" to initialize the label embedding matrix?

Reply:
Yes, this will save the effort of training a linear model during bootstrapping and still have faster convergence than random initialization.